scripts/docs/parse_assets.py

```diff
@@ -31,34 +31,85 @@ def parse_assets(input_dirs: List[Path],
     """
     asset_count = 0
 
-    references = defaultdict(list)
+    references = {}
 
     for asset_config in util.find_assets(input_dirs, asset_config_filename, pattern=pattern):
         asset_count += 1
 
         asset_type, asset_name, asset_file_name, asset_description = \
             generate_asset_documentation.create_asset_doc(asset_config, asset_config.type)
 
-        references[asset_type].append((asset_name, asset_file_name, asset_description))
+        if asset_type not in references:
+            references[asset_type] = defaultdict(list)
+
+        for category in asset_config.categories:
+            references[asset_type][category].append((asset_name, asset_file_name, asset_description))
+
+        # If no categories are found, put the asset under "Uncategorized"
+        if len(asset_config.categories) == 0:
+            references[asset_type]["Uncategorized"].append((asset_name, asset_file_name, asset_description))
+
+        references[asset_type]["All"].append((asset_name, asset_file_name, asset_description))
 
     logger.print(f"{asset_count} asset(s) parsed")
 
     for asset_type in references:
+
+        category_docs_links_list = []
+
+        for category in references[asset_type]:
+            if category == "All":
+                continue
+
+            category_doc_file_name = f"{asset_type}s-{category.replace(' ', '-')}-documentation"
+            capitalized_category = category[0].upper() + category[1:]
+
+            # Add to unordered list of categories
+            category_docs_links_list.append(
+                snakemd.Paragraph(capitalized_category).insert_link(capitalized_category, category_doc_file_name))
+
+            # Create a new markdown file for each category
+            category_doc = snakemd.new_doc()
+            category_doc.add_heading(f"{capitalized_category} {asset_type}s", level=1)
+
+            # Create list of assets under the category
+            category_asset_links_list = []
+
+            for asset_name, asset_file_name, asset_description in references[asset_type][category]:
+                category_asset_links_list.append(
+                    snakemd.Paragraph(asset_name).insert_link(asset_name, category_doc_file_name))
+
+            category_doc.add_unordered_list(category_asset_links_list)
+
+            # Write to category doc
+            with open(f"{asset_type}s/{category_doc_file_name}.md", 'w') as f:
+                f.write(str(category_doc))
+
         # Create a new markdown file for each asset type
         doc = snakemd.new_doc()
-        doc.add_heading(asset_type.capitalize() + "s", level=1)
+        doc.add_heading(f"{asset_type.capitalize()}s", level=1)
+
+        # Add list of asset categories
+        doc.add_heading("Components by categories", level=2)
```

Comment:
"Component" is a specific asset type, so let's avoid that when referring to a assets in general. How about this?
```suggestion
        doc.add_heading("By category", level=2)
```